pkg/util/node.go

```diff
@@ -28,7 +28,7 @@ func GetHostname(hostnameOverride string) string {
 	if string(hostname) == "" {
 		// Note: We use exec here instead of os.Hostname() because we
 		// want the FQDN, and this is the easiest way to get it.
```

Comment:
I think you are right on os.HostName(). uname.2.html says:

       Part of the utsname information is also accessible via
       /proc/sys/kernel/{ostype, hostname, osrelease, version, domainname}.